### PR TITLE
Remove extraneous reagent istype checks, allow carbons (MONKEYS) to get drunk

### PR DESCRIPTION
**Pull request by vuonojenmustaturska**
:cl: Naksu
code: Fixed the signatures of several reagent procs, removing useless checks
tweak: drunkenness is now handled at the carbon level, allowing monkeys to experience ethanol
balance: initropidil also works on carbon level, allowing syndicate operatives using the poison kit to achieve the silent assassin rating against monkeys
/:cl:

### CODE REVIEW

**Comment by vuonojenmustaturska** [comment]
tested in-game on 1427, seems to work as it should

**Comment by PKPenguin321** [comment]
Drunk xenos?

**Comment by Anonmare** [comment]
The new meta strat - alcohol smoke machine to debilitate xenos

**Comment by vuonojenmustaturska** [comment]
testmerged for several rounds, no issues that i can see

File: code/modules/reagents/chemistry/reagents/blob_reagents.dm [hunk 1]
```diff
@@ -199,11 +199,9 @@
...
 	M.apply_damage(0.7*reac_volume, TOX)
 
-/datum/reagent/blob/regenerative_materia/on_mob_life(mob/living/M)
-	M.adjustToxLoss(1*REM)
-	if(iscarbon(M))
-		var/mob/living/carbon/N = M
-		N.hal_screwyhud = SCREWYHUD_HEALTHY //fully healed, honest
+/datum/reagent/blob/regenerative_materia/on_mob_life(mob/living/carbon/C)
+	C.adjustToxLoss(1*REM)
+	C.hal_screwyhud = SCREWYHUD_HEALTHY //fully healed, honest
```
Comment (AnturK) [inline code/modules/reagents/chemistry/reagents/blob_reagents.dm]: Does this actually do anything for non-humans ? Not really relevant just making sure it doesn't break updates completely.

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm [hunk 2]
```diff
@@ -252,27 +248,23 @@
...
-		var/mob/living/L = M
-		L.add_trait(TRAIT_STUNIMMUNE, id)
-		L.add_trait(TRAIT_SLEEPIMMUNE, id)
-		if(iscarbon(L))
-			var/mob/living/carbon/C = L
-			rage = new()
-			C.gain_trauma(rage, TRAUMA_RESILIENCE_ABSOLUTE)
+	L.add_trait(TRAIT_STUNIMMUNE, id)
+	L.add_trait(TRAIT_SLEEPIMMUNE, id)
+	if(iscarbon(L))
```
Comment (AnturK) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm]: Might as well get rid of this ?

**Review by AnturK — COMMENTED** [review]
Just two minor questions

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm (hunk 2 above)
Comment (vuonojenmustaturska) [reply code/modules/reagents/chemistry/reagents/drug_reagents.dm]: `on_mob_add()` works on the living, not just carbons

File: code/modules/reagents/chemistry/reagents/blob_reagents.dm (hunk 1 above)
Comment (vuonojenmustaturska) [reply code/modules/reagents/chemistry/reagents/blob_reagents.dm]: no idea, but it's the same code, it has always run for everyone capable of getting this proc called on them

**Pull request merged**